crates/engine/tree/benches/multiproof_chunking.rs

```diff
@@ -0,0 +1,87 @@
+#![allow(missing_docs)]
+
+use alloy_primitives::{map::B256Set, B256};
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use reth_engine_tree::tree::{select_dynamic_chunk_size, WorkerSnapshot};
+use reth_trie::MultiProofTargets;
+
+const DEFAULT_BASE_CHUNK: usize = 10;
+
+fn make_targets(accounts: usize, slots: usize) -> MultiProofTargets {
+    let mut targets = MultiProofTargets::default();
+    for account in 0..accounts {
+        let hashed = make_hash(account as u64);
+        let mut storage = B256Set::default();
+        for slot in 0..slots {
+            storage.insert(make_hash((account * slots + slot) as u64));
+        }
+        targets.insert(hashed, storage);
+    }
+    targets
+}
+
+fn make_hash(value: u64) -> B256 {
+    let mut bytes = [0u8; 32];
+    bytes[24..].copy_from_slice(&value.to_be_bytes());
+    B256::from(bytes)
+}
+
+fn count_chunks(targets: &MultiProofTargets, chunk_size: usize) -> usize {
+    targets.clone().chunks(chunk_size).count()
+}
+
+/// Benchmarks how many chunks the dynamic policy emits for different idle worker counts.
+fn bench_chunking(c: &mut Criterion) {
```

Comment:
I'm curious what these benchmarks are intended to do/show? How should we use the results from them?